### PR TITLE
filters by file or dataset for listeners

### PR DESCRIPTION
**Pull request by tcnichol**
Changes made in both the front end and back end.

The query to fetchListeners now sends in a field 'process.' The aggregated pipeline query on the back end now checks if it is looking for 'file' or 'dataset' extractors. This is reflected in the analysis tab on the Dataset or File page.

### CODE REVIEW

File: frontend/src/components/listeners/Listeners.tsx [hunk 1]
```diff
@@ -100,14 +103,14 @@ export function Listeners(props: ListenerProps) {
 		setSelectedCategory("");
 
 		if (searchText !== "") searchListeners(searchText, 0, limit, 0);
-		else listListeners(0, limit, 0, selectedCategory, selectedLabel, aliveOnly);
+		else listListeners(0, limit, 0, selectedCategory, selectedLabel, aliveOnly, process);
```
Comment (longshuicy) [inline frontend/src/components/listeners/Listeners.tsx:106]: Some how my search on dataset still show file extractors
image

**Review by longshuicy — CHANGES_REQUESTED** [review]

File: frontend/src/components/listeners/Listeners.tsx (hunk 1 above)
Comment (tcnichol) [reply frontend/src/components/listeners/Listeners.tsx:106]: I'll take a look at the search endpoint when I'm back Friday

Comment (tcnichol) [reply frontend/src/components/listeners/Listeners.tsx:107]: I think this is fixed now, can you check again?

**Review by longshuicy — APPROVED** [review]
works well approve.

**Review by lmarini — APPROVED** [review]

**Pull request merged**